pinot-controller/src/main/java/org/apache/pinot/controller/cursors/ResponseStoreCleaner.java

```diff
@@ -0,0 +1,222 @@
+            String.format("ResponseStore delete response - Broker: %s. Response: %s", key, value)));
+      }
+    } catch (Exception e) {
+      LOGGER.error(e.getMessage());
+    }
+  }
+
+  private Map<String, List<CursorResponseNative>> getAllQueryResults(Map<String, InstanceInfo> brokers,
+      Map<String, String> requestHeaders)
+      throws Exception {
+    String protocol = _controllerConf.getControllerBrokerProtocol();
+    int portOverride = _controllerConf.getControllerBrokerPortOverride();
+    List<String> brokerUrls = new ArrayList<>();
+    for (InstanceInfo broker : brokers.values()) {
+      int port = portOverride > 0 ? portOverride : broker.getPort();
+      brokerUrls.add(String.format(QUERY_RESULT_STORE, protocol, broker.getHost(), port));
+    }
+    LOGGER.debug("Getting running queries via broker urls: {}", brokerUrls);
+    Map<String, String> strResponseMap = getResponseMap(requestHeaders, brokerUrls, "GET", HttpGet::new);
+    return strResponseMap.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> {
+      try {
+        return JsonUtils.stringToObject(e.getValue(), new TypeReference<>() {
+        });
+      } catch (IOException ex) {
+        throw new RuntimeException(ex);
+      }
+    }));
+  }
+
+  private <T extends HttpUriRequestBase> Map<String, String> getResponseMap(Map<String, String> requestHeaders,
+      List<String> brokerUrls, String methodName, Function<String, T> httpRequestBaseSupplier)
+      throws Exception {
+    List<Pair<String, String>> urlsAndRequestBodies = new ArrayList<>(brokerUrls.size());
+    brokerUrls.forEach((url) -> urlsAndRequestBodies.add(Pair.of(url, "")));
+
+    CompletionService<MultiHttpRequestResponse> completionService =
+        new MultiHttpRequest(_executor, _connectionManager).execute(urlsAndRequestBodies, requestHeaders,
+            ResponseStoreCleaner.TIMEOUT_MS, methodName, httpRequestBaseSupplier);
+    Map<String, String> responseMap = new HashMap<>();
+    List<String> errMessages = new ArrayList<>(brokerUrls.size());
+    for (int i = 0; i < brokerUrls.size(); i++) {
+      try (MultiHttpRequestResponse httpRequestResponse = completionService.take().get()) {
+        // The completion order is different from brokerUrls, thus use uri in the response.
+        URI uri = httpRequestResponse.getURI();
+        int status = httpRequestResponse.getResponse().getCode();
+        String responseString = EntityUtils.toString(httpRequestResponse.getResponse().getEntity());
+        // Unexpected server responses are collected and returned as exception.
+        if (status != 200) {
+          throw new Exception(
+              String.format("Unexpected status=%d and response='%s' from uri='%s'", status, responseString, uri));
+        }
+        responseMap.put((getInstanceKey(uri.getHost(), Integer.toString(uri.getPort()))), responseString);
+      } catch (Exception e) {
+        LOGGER.error("Failed to get queries", e);
```

Comment:
This method is also being used to delete responses right?

Reply:
For GET and DELETE. I have used `methodName` in the log.